pkg/extension/opampextension/opamp_agent.go

```diff
@@ -424,6 +425,18 @@ func (o *opampAgent) saveEffectiveConfig(dir string) error {
 		if err != nil {
 			return err
 		}
+		o.logger.Debug("Loading Component Factories...")
+		factories, err := Components()
+		if err != nil {
+			return fmt.Errorf("cannot get the list of factories: %v", err)
+		}
+		o.logger.Info("Loading Configuration to Validate...")
+		_, errValidate := otelcoltest.LoadConfigAndValidate(p, factories)
+		if errValidate != nil {
+			o.logger.Error("Validation Failed... %v", zap.Error(errValidate))
+			return fmt.Errorf("cannot validate config named %v", errValidate)
```

Comment:
Is this error formatting correct? Seems like it should be something like `Errorf("cannot validate config named %q: %s", name, err)`

Reply:
This is a test which shows the example formatting
https://github.com/SumoLogic/sumologic-otel-collector/pull/1469/files#diff-8c8da29ce13818d83c803112ef661f02f923afe91f5c30f120447ae2204d896fR240

Reply:
@echlebek does the above example make sense or would you like to see a change in the format?

Reply:
I see, that makes sense